File: grpc/examples/go/get-order-by-reference.go
package main

import (
	"context"
	"fmt"

	"github.com/vegaprotocol/api/grpc/clients/go/generated/code.vegaprotocol.io/vega/proto/api"
	"github.com/vegaprotocol/api/grpc/examples/go/helpers"

	"google.golang.org/grpc"
)

func main() {
	nodeURLGrpc := helpers.GetFromEnv("NODE_URL_GRPC")
	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	dataClient := api.NewTradingDataServiceClient(conn)

	// __get_order_by_ref:
	// Request an order by reference on a Vega network
	// Note: This is an example and order reference will be provided in the response
	// from a prepareSubmitOrder request in the field named `submitID` or similar.
	reference := "4617844f-6fab-4cf6-8852-e29dbd96e5f1"
	request := api.OrderByReferenceRequest{Reference: reference}
	order, err := dataClient.OrderByReference(context.Background(), &request)
	if err != nil {
		panic(err)
	}

	fmt.Printf("OrderByReference: %s", order)
	// :get_order_by_ref__
}

File: grpc/examples/python/get-order-by-reference.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega node (gRPC)

Apps/Libraries:
- gRPC: Vega-API-client (https://pypi.org/project/Vega-API-client/)
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import helpers

# __import_client:
import vegaapiclient as vac
# :import_client__

node_url_grpc = helpers.get_from_env("NODE_URL_GRPC")

# __get_order_by_ref:
# Request an order by reference on a Vega network
# Note: This is an example and order reference will be provided in the response
# from a prepareSubmitOrder request in the field named `submitID` or similar.
reference = "15b757d2-9dcd-4442-81da-18c709c8d97e"
data_client = vac.VegaTradingDataClient(node_url_grpc)
order_by_ref_request = vac.api.trading.OrderByReferenceRequest(
    reference=reference
)
response = data_client.OrderByReference(order_by_ref_request)
print("OrderByReference:\n{}".format(response))
# :get_order_by_ref__
